Framework/script/RepoCleaner/qcrepocleaner/rules/multiple_per_run.py

```diff
@@ -0,0 +1,130 @@
+from datetime import datetime
+from datetime import timedelta
+import logging
+from typing import Dict, DefaultDict, List
+
+from qcrepocleaner.Ccdb import Ccdb, ObjectVersion
+from qcrepocleaner.policies_utils import in_grace_period, group_versions
+
+from qcrepocleaner import policies_utils
+
+logger = logging  # default logger
+
+def process(ccdb: Ccdb, object_path: str, delay: int,  from_timestamp: int, to_timestamp: int,
+            extra_params: Dict[str, str]):
+    '''
+    Process this deletion rule on the object. We use the CCDB passed by argument.
+    Objects who have been created recently are spared (delay is expressed in minutes).
+
+    This specific policy, multiple_per_run, operates like this : Keep the first and the last version of a run plus
+    1 version every X minutes.
+
+    Extra parameters:
+      - migrate_to_EOS: Migrate the . (default: false)
+      - interval_between_versions: Period in minutes between the versions we will keep. (default: 90)
+
+    It is implemented like this :
+        Map of buckets: run[+pass+period] -> list of versions
+        Go through all objects: Add the object to the corresponding key (run[+pass+period])
+        Remove the empty run from the map (we ignore objects without a run)
+        Go through the map: for each run (resp. run+pass+period)
+            Get SOR (validity of first object)
+            if SOR < now - delay
+                do
+                    keep first
+                    delete everything for the next interval_between_versions
+                until no more
+                move the last one, from delete to preserve
+
+    :param ccdb: the ccdb in which objects are cleaned up.
+    :param object_path: path to the object, or pattern, to which a rule will apply.
+    :param delay: delay since SOR after which we process the run altogether.
+    :param from_timestamp: only objects created after this timestamp are considered.
+    :param to_timestamp: only objects created before this timestamp are considered.
+    :param extra_params: a dictionary containing extra parameters for this rule.
+    :return a dictionary with the number of deleted, preserved and updated versions. Total = deleted+preserved.
+    '''
+    
+    logger.debug(f"Plugin multiple_per_run processing {object_path}")
+
+    preservation_list: List[ObjectVersion] = []
+    deletion_list: List[ObjectVersion] = []
+    update_list: List[ObjectVersion] = []
+    versions_buckets_dict: DefaultDict[str, List[ObjectVersion]] = defaultdict(list)
+    metadata_for_preservation = {'preservation': 'true'}
+
+    # config parameters
+    period_pass = (extra_params.get("period_pass", False) == True)
+    logger.debug(f"period_pass : {period_pass}")
+    interval_between_versions = int(extra_params.get("interval_between_versions", 30))
+    logger.debug(f"interval_between_versions : {interval_between_versions}")
+    migrate_to_EOS = (extra_params.get("migrate_to_EOS", False) is True)
```

Comment:
Is there a reason why you compare `period_pass` using `== True`, but `migrate_to_EOS` with `is True`?

Reply:
good point. Modern Python dictates to use `is True`. I will change the other one.